utils/wrappers/aws/api_gateway.py

```diff
@@ -41,11 +42,13 @@ def wrapper(event: dict, context: LambdaContext, *args: Any, **kwargs: Any) -> d
                 'not_found': 404
             }
 
-            headers = {}
+            requestHeaders = event.get('headers', {})
+            origin = requestHeaders.get('origin', requestHeaders.get('Origin', None))
 
-            if CORS_ALLOWED_ORIGIN:
-                headers = {
-                    'Access-Control-Allow-Origin': CORS_ALLOWED_ORIGIN,
+            responseHeaders = {}
+            if origin is not None and re.match(CORS_ALLOWED_REGEX, origin):
```

Comment:
Do we use any requests that do not include the `Origin` header? And not having cors headers for not allowed origins work as intended?

Also, should we not run the handler if the origin does not match?

Reply:
Do we use any requests that do not include the Origin header? And not having cors headers for not allowed origins work as intended?

- I don't think we have those requests. Only node_data_aggregator handler does not have the ApiGateway() decorator.

Also, should we not run the handler if the origin does not match?

- You mean ,we should not do anything if CORS does not match?